src/main/java/dan200/computercraft/core/apis/TermAPI.java

```diff
@@ -298,6 +299,31 @@ public static void setColour( Terminal terminal, int colour, double r, double g,
                 }
                 return null;
             }
+            case 23:
+            {
+                // setPixel
+                int x = getInt( args, 0 ) - 1;
+                int y = getInt( args, 1 ) - 1;
+                String text = optString( args, 2, null );
+                String textColour = optString( args, 3, null );
+                String backgroundColour = optString( args, 4, null );
+                synchronized( m_terminal )
+                {
+                    if ( x <= 0 || x >= m_terminal.getWidth() || y <= 0 || y >= m_terminal.getHeight() ) {
+                        throw new LuaException( "Position out of range" );
+                    }
+                    if ( text != null ) {
+                        m_terminal.setPixelChar( x, y, text.substring( 0, 1) );
```

Comment:
Will these not fail if the string is empty? Also, seeing as we're just writing a single character, could the `Terminal.*` functions accept a `char` instead?